src/assignments/phat/lesson_3_answers.clj

```diff
@@ -0,0 +1,143 @@
+(ns lesson-3-answers.core)
+(require '[clojure.string :as str])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; 
+;; 128 - Recognize Playing Cards
+
+;; define maps for suit and rank
+;; split input string into 2 parts
+;; first part is a key in suit
+;; second part is a key in rank
+
+(defn recognize-card
+  "Take a string and convert into a map of suit and rank"
+  [card]
+  (let [suit {"D" :diamond 
+              "H" :heart 
+              "C" :club 
+              "S" :spade}
+        card-rank {"2" 0 "3" 1 "4" 2 "5" 3 "6" 4 
+                   "7" 5 "8" 6 "9" 7 "T" 8 "J" 9
+                   "Q" 10 "K" 11 "A" 12}]
+    (hash-map :suit (suit (first (str/split card #"")))
+              :rank (card-rank (last (str/split card #""))))))
+
+;; tests
+(= {:suit :diamond :rank 10} (recognize-card "DQ"))
+(= {:suit :heart :rank 3} (recognize-card "H5"))
+(= {:suit :club :rank 12} (recognize-card "CA"))
+(= (range 13) (map (comp :rank recognize-card str)
+                   '[S2 S3 S4 S5 S6 S7
+                     S8 S9 ST SJ SQ SK SA]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; 178 - Best hand
+
+(defn sort-cards
+  "Recognize all cards and group cards by suit or rank"
+  [card-coll keyword]
+  (into (sorted-map) (group-by keyword (map recognize-card card-coll))))
+
+
+(defn continuous-sequence?
+  "check if a sequence increases with a step of 1"
+  [coll]
+  (apply = 1 (map - (rest coll) coll)))
```

Comment:
Lovely.  But be aware that if coll grew very large, this could be very slow.  (This case here is fine, of course).

Check out the function every?